.github/workflows/pull_request.yml

```diff
@@ -22,13 +22,23 @@ jobs:
       - uses: actions/setup-python@v4
         with:
           python-version: ${{ env.PYTHON_VERSION }}
-          cache: 'pipenv'
-      - name: Install Pipenv
-        run: pip install pipenv==2023.8.22
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
+        with:
+          version: 1.7.1
+          virtualenvs-create: true
+          virtualenvs-in-project: false
+          virtualenvs-path: ~/.virtualenvs
+      - name: Cache Poetry virtualenv
+        uses: actions/cache@v3
+        id: cache-virtualenv
+        with:
+          path: ~/.virtualenvs
+          key: ${{ runner.os }}-${{ env.PYTHON_VERSION }}-virtualenvs-${{ hashFiles('poetry.lock') }}
       - name: Install virtual environment
         run: |
           sudo apt-get install libsnappy-dev
-          pipenv install --dev
+          poetry install --no-interaction 
```

Comment:
Poetry by default installs dev dependencies.  Use `--only main` to install the main deps only. Need to change references elsewhere also.

Reply:
So here, since it was installed the dev dependencies, this reference is correct.

Reply:
Oh I thought it was just good practice because of examples I saw on the docs. Now Removed 👍